.github/workflows/main.yml

```diff
@@ -1,29 +1,25 @@
 name: main
+
 on:
   push:
     branches:
-      - master
       - cluster
-    paths-ignore:
-      - "docs/**"
-      - "**.md"
-      - "dashboards/**"
-      - "deployment/**.yml"
+      - master
+    paths:
+      - '**.go'
   pull_request:
     branches:
-      - master
       - cluster
-    paths-ignore:
-      - "docs/**"
-      - "**.md"
-      - "dashboards/**"
-      - "deployment/**.yml"
+      - master
+    paths:
+      - '**.go'
+
 permissions:
   contents: read
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
 
 jobs:
   lint:
```

Comment:
Is it possible to do lint and tests in one job? This would help to skip Go install phase

Reply:
I don't think this is true. See the following action run as an example: https://github.com/VictoriaMetrics/VictoriaMetrics/actions/runs/9746369185
Here is a graph of that run:
![1719912664](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/1367798/ca793408-caa5-4dbb-8ed0-f1b8b8f898d0)

The idea is that each step is independent of each other. Github uses `needs` in order to build a dependency graph and executes each step once.

Reply:
My understanding is that lint and test(matrix) on the screenshot's run are 2 separate jobs. @hagen1778 asked about making it a single one in order to skip Go install phase.

Just want to make sure we're on the same page @zekker6